src/storage-client/src/sink.rs

```diff
@@ -231,34 +233,95 @@ pub async fn publish_kafka_schemas(
     topic: String,
     key_schema: Option<String>,
     key_schema_type: Option<mz_ccsr::SchemaType>,
-    value_schema: &str,
+    value_schema: String,
     value_schema_type: mz_ccsr::SchemaType,
+    key_compatibility_level: Option<mz_ccsr::CompatibilityLevel>,
+    value_compatibility_level: Option<mz_ccsr::CompatibilityLevel>,
 ) -> Result<(Option<i32>, i32), anyhow::Error> {
-    let value_schema_id = ccsr
-        .publish_schema(
-            &format!("{}-value", topic),
-            value_schema,
-            value_schema_type,
-            &[],
-        )
+    let key_subject = format!("{}-key", topic);
+    let value_subject = format!("{}-value", topic);
+
+    if let Some(key_compatibility_level) = key_compatibility_level {
+        let ccsr = ccsr.clone();
+        let key_subject = key_subject.clone();
+        async move {
+            // Only update the compatibility level if it's not already set to something.
+            match ccsr.get_subject_config(&key_subject).await {
+                Ok(config) => {
+                    if config.compatibility_level != key_compatibility_level {
+                        tracing::debug!(
+                            "key compatibility level '{}' does not match intended '{}'",
+                            config.compatibility_level,
+                            key_compatibility_level
+                        );
+                    }
+                    Ok(())
+                }
+                Err(GetSubjectConfigError::SubjectCompatibilityLevelNotSet) => ccsr
```

Comment:
(You checked for `SubjectNotFound` for the value schema below but not here for the key schema!)